NDTensors/src/UnallocatedArrays/src/unallocatedfill.jl

```diff
@@ -0,0 +1,34 @@
+## TODO All constructors not fully implemented but so far it matches the 
+## constructors found in `FillArrays`. Need to fix io
+struct UnallocatedFill{ElT,N,Axes,Alloc<:AbstractArray} <:
+       FillArrays.AbstractFill{ElT,N,Axes}
+  f::FillArrays.Fill{ElT,N,Axes}
+
+  function UnallocatedFill{ElT,N,Axes,Alloc}(x::ElT, inds::Tuple) where {ElT,N,Axes,Alloc}
+    f = FillArrays.Fill(x, inds)
+    ax = typeof(FillArrays.axes(f))
+    return new{ElT,N,ax,Alloc}(f)
+  end
+
+  function UnallocatedFill{ElT,0,Tuple{},Alloc}(x::ElT, inds::Tuple{}) where {ElT,Alloc}
+    f = FillArrays.Fill{ElT,0,Tuple{}}(x, inds)
+    return new{ElT,0,Tuple{},Alloc}(f)
+  end
```

Comment:
Isn't this version covered by the version above? Also, I don't really see why we need constructors that take the element type as an argument, since you can specify it using a type parameter.

Reply:
Hmm I am not sure if this is covered by the original constructor I took the constructors from what is in [`FillArrays`](https://github.com/JuliaArrays/FillArrays.jl/blob/7036a5005d2a96f79b67ffc9bfd4a1e07517c85e/src/FillArrays.jl#L107C1-L114C4). I just removed this constructor and I am able to make a empty `UnallocatedFill` using the first constructor

Reply:
Hmm I am not sure if this is covered by the original constructor I took the constructors from what is in [`FillArrays`](https://github.com/JuliaArrays/FillArrays.jl/blob/7036a5005d2a96f79b67ffc9bfd4a1e07517c85e/src/FillArrays.jl#L107C1-L114C4). I just removed this constructor and I am able to make a empty `UnallocatedFill` using the first constructor